### ISSUE TITLE
Create Quiz type

### ISSUE THREAD

**Issue by lucy-kind** [schemaorg/schemaorg#2611]
Create Quiz as subtype of LearningResource. Inherits all properties; no new properties at this time.

**Comment by danbri** [comment]
Context - This is a proposal from Google based on our experience with actively consuming schema.org edu-related markup. If it were accepted, it would make it easier for us and others to understand when a piece of web content is some kind of Quiz.

**Comment by Aaranged** [comment]
Quick thought on this is that a quiz can be for entertainment as well as educational purposes. That is, while there's obviously value in modeling this Thing, I think it might cause future problems if a quiz were restricted to the LearningResource domain (indeed, [Google itself](https://developers.google.com/assistant/templates/trivia) promotes the creation "quizzes" that are not necessarily "educational").

**Comment by lifan0127** [comment]
Reference #1401 for context. I understand that LearningResource has an explicit orientation for education.

> Quick thought on this is that a quiz can be for entertainment as well as educational purposes. That is, while there's obviously value in modeling this Thing, I think it might cause future problems if a quiz were restricted to the LearningResource domain (indeed, [Google itself](https://developers.google.com/assistant/templates/trivia) promotes the creation "quizzes" that are not necessarily "educational").

**Comment by philbarker** [comment]
The name Quiz and definition suggested in the PR (Quiz: A test of knowledge.) does look more like the entertainment type of quiz. No matter what it is a subtype of, I think you will see a lot of people using it for [pub quizzes](https://en.wikipedia.org/wiki/Pub_quiz) and the like.

The definition ought at least cover assessment of skils and abilities as well as knowledge.

**Comment by danbri** [comment]
@philbarker the current draft definition is "Quiz: A test of knowledge."; how about "Quiz: A test of knowledge, skills and abilities."? /cc @lucy-kind 

@Aaranged I think we can take a somewhat broad view of educational vs entertainment purposes. If someone makes a quiz which is "Dan's Quiz - learn what your spirit animal is by sharing your mother's maiden name and date of birth!", ... the fact that this isn't really very educational isn't the real problem. There will be good and bad LearningResource-tagged content, as with everything else. Maybe there will be some quizzes that are only tenuously educational but I think we can deal with that problem if it proves to be one.

**Comment by philbarker** [comment]
@danbri Yup that's better.  Are you set on the name "Quiz" rather than "Assessment" or "Test". The boring more edu-speak Assessment might help avoid some of the more trival quizzes. If you want to avoid any suggestion that exams are included, then "Formative" in the name and definition would do it. But not all assessments are quiz-like (think driving test). Naming things is hard, who knew?

**Comment by danbri** [comment]
I think "Quiz" works for now. Formative is a bit too much of a technical term for us here, I think. 

Our proposal for a type is coming from a more educational angle but SEOs will see this and immediately wonder whether content like https://www.capitalfm.com/features/quizzes/one-direction-tattoo-trivia-quiz/ is also a Quiz. I would like to be able to answer "yeah, just a kinda trashy / fun one". Is it a LearningResource? Barely, but since it does evaluate one's knowledge of a particular domain, I could defend the claim that it was. Rather than being snooty about whether things are learning-oriented enough to deserve being called a LearningResource I think we'll get more value from using other (LRMI etc.) properties to help distinguish the different kinds of learning resources (quizzes or whatever) that are out there... Does that perspective work for you folks?

**Comment by philbarker** [comment]
@danbri I'll check with other folk in LRMI, but like you, I think Quiz works for now. We can always build in other assessment terms around it if/when we need them. There will always be edge cases and different perspectives about quality for what is a learning resource.

**Comment by jgoodell2** [comment]
I think this would be a valuable addition to LRMI & Schema. We should consider whether the use case also includes formal assessments (test, exam) in addition to informal checks for understanding (quiz). The ceds.ed.gov [Assessment domain](https://ceds.ed.gov/domainEntitySchema.aspx?v=8&txt=43390) has a plethora of terms that apply to all kinds of assessments and assessment items, and data used to manage delivery of assessments and assessment results...too many concepts for practical use by Schema.org, but it gives a catalog of concepts and properties to consider.

**Comment by science** [comment]
Since we have general consensus that schema.org models educational things that "teach" or things that "measure," having an explicit type for the "measure" side of things is useful. Quiz seems like an accessible term for this type, at the level of schema.org. 

I'll be interested in potentially subclassing/profiling this in the OCX schema to include additional metadata around measurement, so this is a useful type for that type of use-case also. 

+1

**Comment by testometrika** [comment]
The type Testing or Quiz is sorely lacking for online tests. We specialize in tests https://en.testometrika.com and can provide advice.

We are currently using an inappropriate CreativeWork, but there is a problem with AggregateRating. Invalid target type for itemReviewed property.

The test can contain title, description, rating, author, datePublished, dateModified, list of questions (if not ajax) with answer options .

**Comment by psychology-tools** [comment]
We've added a first-draft implementation of the pending `Quiz` schema to all the multiple choice self-assessments on our site (e.g. [https://psychology-tools.com/test/gad-7](https://psychology-tools.com/test/gad-7). Combined with the `Question` and `Answer` (as an array of `suggestedAnswer` types). Just wanted to provide an _in-the-wild_ example and say this is a really useful schema!

**Comment by macgyver** [comment]
I work at a media organization that does a quiz (more entertainment than assessment) with ~28k readers each week, what is the incentive for me to implement the Quiz schema?

**Comment by westcoast42** [comment]
Founder of FunTrivia.com here.   We host 150,000+ community sourced and maintained quizzes across 25,000 topics.   Everything from particle physics to Seinfeld.   We totally support the addition of LearningResource:Quiz.   Right now we are using Article to describe our quizzes, but clearly Quiz would be more accurate.  

I suggest Question/Answer be included as optional sub-elements of this type.

**Comment by psychology-tools** [comment]
> Founder of FunTrivia.com here.   We host 150,000+ community sourced and maintained quizzes across 25,000 topics.   Everything from particle physics to Seinfeld.   We totally support the addition of LearningResource:Quiz.   Right now we are using Article to describe our quizzes, but clearly Quiz would be more accurate.  
> 
> 
> 
> I suggest Question/Answer be included as optional sub-elements of this type.

I think the Quiz/Question/Answer structure works well for both "Learning Resource" type Quizzes. 

It works just as well for the "Subjective Assessment" type of Quizzes (with the exception of a required "expectedAnswer" field, which can be kludged by expecting all answers). 

I would include in the "Subjective Assessment" category both highly serious, rigorously scored, psychological self-assessment tools and informal "what flavor of dinosaur" buzz-feed type quizzes.

It appears to me that the LearningResource/Quiz and the above described "Subjective Assessment" Quiz, although being almost identical in their structure, are very different in their contexts and purposes. And I would venture to guess that the two types are about equally prevalent in the wild.

The only two ways I can see to resolve this would be either:

# Option 1
Make **Quiz** a direct subtype of **CreativeWork** and have two subtypes like **ObjectiveQuiz** and **SubjectiveQuiz** or **LearningResourceQuiz** and **AssessmentResourceQuiz**. 

This would have the benefit of allowing both types of quiz to inherit mostly all of their properties, and keeping their mostly identical schemas in line with each other in future.

Furthermore, per the specification  **LearningResource** is "expected to be used as an addition to a primary type such as Book, VideoObject, Product etc." So it is unclear to me why **LearningResource** would have any direct subtypes given this.


# Option 2
Create a separate type of quiz for the non-learning resource type. It would have to be called something other than **Quiz**, and could not be a subtype of **Quiz**. 

Excluding roughly half of the objects commonly called "quiz" with __nearly idential structure and properties__ from the **Quiz** schema based on a single point of distinction (educational/subjective) does not seem like the cleanest solution.

**Comment by psychology-tools** [comment]
The Course type (the only other subtype of LearningResource) is both a primary direct subtype of CreativeWork and a subtype of LearningResource.

Wouldn't treating Quiz the same way as its only sibling subtype resolve this issue?

Wouldn't either eliminating all subtypes of LearningResource also solve this problem?

**Comment by westcoast42** [comment]
It might be most straightforward to just have an attribute under “Quiz” to differentiate the type of quiz.   
QuizType : text:  “Factual”, “Personality”, Etc or something.

Definitely worth disambiguating though because a “What Harry Potter character are you?” Quiz And a “French Revolution” Quiz are worlds apart.

**Comment by webshark** [comment]
Has any progress been made for this Schema ? Any online examples I could look at?
